sim-lib/src/lib.rs

```diff
@@ -62,27 +62,81 @@ impl NodeId {
         }
         Ok(())
     }
+
+    pub fn get_pk(&self) -> Result<&PublicKey, String> {
+        if let NodeId::PublicKey(pk) = self {
+            Ok(pk)
+        } else {
+            Err("NodeId is not a PublicKey".to_string())
+        }
+    }
+}
+
+impl std::fmt::Display for NodeId {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        write!(
+            f,
+            "{}",
+            match self {
+                NodeId::PublicKey(pk) => pk.to_string(),
+                NodeId::Alias(a) => a.to_owned(),
+            }
+        )
+    }
 }
 
 #[derive(Debug, Serialize, Deserialize, Clone)]
 pub struct Config {
     pub nodes: Vec<NodeConnection>,
-    pub activity: Option<Vec<ActivityDefinition>>,
+    #[serde(default)]
+    pub activity: Vec<ActivityParser>,
 }
 
-#[derive(Debug, Clone, Copy, Serialize, Deserialize)]
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct ActivityParser {
```

Comment:
nit: would be worthwhile to have docs explaining why we have these two structs, I can see this taking some time to figure out as a newcomer to the codebase.

Reply:
Makes sense